### PR TITLE
Implement writev syscall so that musl printf works

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

**Comment by sbc100** [comment]
I'll try to resist the temptation to implement too much more of the syscall interface but being able to use printf here is big benefit IMHO

File: arch/wasm32/wasm.js [hunk 1]
```diff
@@ -24,14 +24,27 @@
...
+var heap_uint32;
+
+function setHeap(h) {
+  heap = h
+  heap_uint8 = new Uint8Array(heap);
+  heap_uint32 = new Uint32Array(heap);
+  heap_size_bytes = heap.byteLength;
+}
+
+setHeap(memory.buffer)
```
Comment (jfbastien) [inline arch/wasm32/wasm.js:38]: This will fail if code ever grows memory. Is that a concern for later? Should probably file an issue.

**Review by jfbastien — APPROVED** [review]

File: arch/wasm32/wasm.js (hunk 1 above)
Comment (sbc100) [reply arch/wasm32/wasm.js:38]: You mean if memory ever grows? 

Yes, although that is true before this change too, no?

Comment (jfbastien) [reply arch/wasm32/wasm.js:38]: Yes.

**Pull request merged**